docs/docs/concepts/functional_api.md

```diff
@@ -155,117 +153,147 @@ You will usually want to pass a **checkpointer** to the `@entrypoint` decorator
+    from langgraph.store.base import BaseStore
+    from langgraph.store.memory import InMemoryStore
     from langchain_core.runnables import RunnableConfig
 
-    @entrypoint(checkpointer=checkpointer)
-    def my_workflow(some_input: dict, config: RunnableConfig) -> int:
-        # some logic that may involve long-running tasks like API calls,
-        # and may be interrupted for human-in-the-loop
-        print(config) # Access the configuration
-        ...
+    in_memory_store = InMemoryStore(...)  # An instance of InMemoryStore for long-term memory
+
+    @entrypoint(
+        checkpointer=checkpointer,  # Specify the checkpointer
+        store=in_memory_store  # Specify the store
+    )  
+    def my_workflow(
+        some_input: dict,  # The input (e.g., passed via `invoke`)
+        *,
+        previous: Any = None, # For short-term memory
+        store: BaseStore,  # For long-term memory
+        writer: StreamWriter,  # For streaming custom data
+        config: RunnableConfig  # For accessing the configuration passed to the entrypoint
+    ) -> ...:
     ```
 
-
-!!! important "Serialization"
-
-    The **inputs** and **outputs** of entrypoints must be JSON-serializable to support checkpointing. Please see the [serialization](#serialization) section for more details.
-
 ### Executing
 
 Using the [`@entrypoint`](#entrypoint) yields a [EntrypointPregel][langgraph.func.EntrypointPregel] object that can be executed using the `invoke`, `ainvoke`, `stream`, and `astream` methods.
 
 === "Invoke"
 
     ```python
-    my_workflow.invoke(some_input)  # Wait for the result synchronously
+    config = {
+        "configurable": {
+            "thread_id": "some_thread_id"
+        }
+    }
+    my_workflow.invoke(some_input, config)  # Wait for the result synchronously
     ```
 
 === "Async Invoke"
 
     ```python
-    await my_workflow.ainvoke(some_input)  # Await result asynchronously
+    config = {
+        "configurable": {
+            "thread_id": "some_thread_id"
+        }
+    }
+    await my_workflow.ainvoke(some_input, config)  # Await result asynchronously
     ```
 
 === "Stream"
     
     ```python
-    for chunk in my_workflow.stream(some_input):
+    config = {
+        "configurable": {
+            "thread_id": "some_thread_id"
+        }
+    }
+
+    for chunk in my_workflow.stream(some_input, config):
         print(chunk)
     ```
 
 === "Async Stream"
 
     ```python
-    async for chunk in my_workflow.astream(some_input):
+    config = {
+        "configurable": {
+            "thread_id": "some_thread_id"
+        }
+    }
+
+    async for chunk in my_workflow.astream(some_input, config):
         print(chunk)
     ```
 
 ### Resuming
 
-Execution can be **resumed** using the [Command][langgraph.types.Command] primitive.
+Resuming an execution after an [interrupt][langgraph.types.interrupt] can be done by passing a **resume** value to the [Command][langgraph.types.Command] primitive.
```

Comment:
should we mention that people should be using non-null resume value?

Reply:
in the API reference

Reply:
too much detail for the conceptual docs I think -- even what we have here is arguably too much detail